### PR TITLE
feat(jdbc): JDBC Outbound connector - Element template creation

### PR DESCRIPTION
**Pull request by johnBgood**
## Description

**First PR of the JDBC Connector Epic.** 

I'll create many, small PRs to avoid a massive PR when my task is done. Keep in mind that some things might change, and this is the point: raising concerns/discussions ASAP in the development process.

- Initial project commit
- New JDBC Outbound templates
- Connector inputs tests

Here's the generated Element Template:
Screenshot 2024-04-05 at 16 33 00

And here's the detailed authentication version:
Screenshot 2024-04-05 at 16 42 42

## Related issues

closes https://github.com/camunda/team-connectors/issues/717

### CODE REVIEW

File: connectors/jdbc/src/main/java/io/camunda/connector/jdbc/model/auth/DetailedAuthentication.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ * except in compliance with the proprietary license.
+ */
+package io.camunda.connector.jdbc.model.auth;
+
+import io.camunda.connector.generator.java.annotation.TemplateProperty;
+import io.camunda.connector.generator.java.annotation.TemplateSubType;
+import jakarta.validation.constraints.NotBlank;
+
+@TemplateSubType(id = "detailed", label = "Detailed")
+public record DetailedAuthentication(
```
Comment (johnBgood) [inline connectors/jdbc/src/main/java/io/camunda/connector/jdbc/model/auth/DetailedAuthentication.java:14]: a better name is more than welcome :)

**Comment by johnBgood** [comment]
@bastiankoerber Please let me know if the UI suits the user's needs 🙏

File: connectors/jdbc/src/main/resources/icon.svg
Comment (johnBgood) [inline connectors/jdbc/src/main/resources/icon.svg:1]: not sure about our process with icons. Should legal by implied? Do we have an icons db?

File: connectors/jdbc/src/main/java/io/camunda/connector/jdbc/model/auth/JdbcAuthentication.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+@JsonSubTypes({
+  @JsonSubTypes.Type(value = UriAuthentication.class, name = "uri"),
+  @JsonSubTypes.Type(value = DetailedAuthentication.class, name = "detailed")
+})
+@TemplateDiscriminatorProperty(
+    label = "Connection type",
+    group = "authentication",
+    name = "authType",
+    defaultValue = "uri")
+public sealed interface JdbcAuthentication permits UriAuthentication, DetailedAuthentication {}
```
Comment (sbuettner) [inline connectors/jdbc/src/main/java/io/camunda/connector/jdbc/model/auth/JdbcAuthentication.java:23]: What about `JdbcConnection` instead as it contains authentication and technical connection details.

File: connectors/jdbc/src/main/java/io/camunda/connector/jdbc/model/auth/DetailedAuthentication.java (hunk 1 above)
Comment (sbuettner) [reply connectors/jdbc/src/main/java/io/camunda/connector/jdbc/model/auth/DetailedAuthentication.java:14]: `CredentialsConnection`?

File: connectors/jdbc/element-templates/hybrid/jdbc-outbound-connector-hybrid.json [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+      "notEmpty" : true
+    },
+    "group" : "database",
+    "binding" : {
+      "name" : "database",
+      "type" : "zeebe:input"
+    },
+    "type" : "Dropdown",
+    "choices" : [ {
+      "name" : "MSSQL",
```
Comment (sbuettner) [inline connectors/jdbc/element-templates/hybrid/jdbc-outbound-connector-hybrid.json:61]:
```suggestion
      "name" : "Microsoft SQL Server",
```

File: connectors/jdbc/element-templates/jdbc-outbound-connector.json [hunk 4]
```diff
@@ -0,0 +1,284 @@
+{
+  "$schema" : "https://unpkg.com/@camunda/zeebe-element-templates-json-schema/resources/schema.json",
+  "name" : "SQL Database Connector",
+  "id" : "io.camunda.connectors.Jdbc.v1",
+  "description" : "Read and write data from a Camunda process directly to a SQL database(MSSQL, MySQL, PostgreSQL)",
```
Comment (sbuettner) [inline connectors/jdbc/element-templates/jdbc-outbound-connector.json:5]:
```suggestion
  "description" : "Read and write data from a Camunda process directly to a SQL database(Microsoft SQL Server, MySQL, PostgreSQL)",
```

**Review by sbuettner — APPROVED** [review]
LGTM

**Review by sbuettner — APPROVED** [review]

**Pull request merged**